drivers/sensor/ccs811/ccs811.c

```diff
@@ -0,0 +1,218 @@
+		 */
+		uval = drv_data->resistance >> 10;
+		val->val1 = uval / 1000000;
+		val->val2 = uval % 1000000;
+
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api ccs811_driver_api = {
+	.sample_fetch = ccs811_sample_fetch,
+	.channel_get = ccs811_channel_get,
+};
+
+static int switch_to_app_mode(struct device *i2c)
+{
+	u8_t status, buf;
+
+	SYS_LOG_DBG("Switching to Application mode...");
+
+	if (i2c_reg_read_byte(i2c, CONFIG_CCS811_I2C_ADDR,
+			      CCS811_REG_STATUS, &status) < 0) {
+		SYS_LOG_ERR("Failed to read Status register");
+		return -EIO;
+	}
+
+	/* Check for the application firmware */
+	if (!(status & CCS811_STATUS_APP_VALID)) {
+		SYS_LOG_ERR("No Application firmware loaded");
+		return -EINVAL;
+	}
+
+	buf = CCS811_REG_APP_START;
+	/* Set the device to application mode */
+	if (i2c_write(i2c, &buf, 1, CONFIG_CCS811_I2C_ADDR) < 0) {
+		SYS_LOG_ERR("Failed to set Application mode");
+		return -EIO;
+	}
+
+	if (i2c_reg_read_byte(i2c, CONFIG_CCS811_I2C_ADDR,
+			      CCS811_REG_STATUS, &status) < 0) {
+		SYS_LOG_ERR("Failed to read Status register");
+		return -EIO;
+	}
+
+	/* Check for application mode */
+	if (!(status & CCS811_STATUS_FW_MODE)) {
+		SYS_LOG_ERR("Failed to start Application firmware");
+		return -EINVAL;
+	}
+
+	SYS_LOG_DBG("CCS811 Application firmware started!");
+
+	return 0;
+}
+
+int ccs811_init(struct device *dev)
+{
+	struct ccs811_data *drv_data = dev->driver_data;
+	int ret;
+	u8_t hw_id;
+
+	drv_data->i2c = device_get_binding(CONFIG_CCS811_I2C_MASTER_DEV_NAME);
+	if (drv_data->i2c == NULL) {
+		SYS_LOG_ERR("Failed to get pointer to %s device!",
+			    CONFIG_CCS811_I2C_MASTER_DEV_NAME);
+		return -EINVAL;
+	}
+
+	/*
+	 * Wakeup pin should be pulled low before initiating any I2C transfer.
+	 * If it has been tied to GND by default, skip this part.
+	 */
+#ifdef CONFIG_CCS811_GPIO_WAKEUP
+	drv_data->gpio = device_get_binding(CONFIG_CCS811_GPIO_DEV_NAME);
+	if (drv_data->gpio == NULL) {
+		SYS_LOG_ERR("Failed to get pointer to %s device!",
+			    CONFIG_CCS811_GPIO_DEV_NAME);
+		return -EINVAL;
+	}
+
+	gpio_pin_configure(drv_data->gpio, CONFIG_CCS811_GPIO_WAKEUP_PIN_NUM,
+			   GPIO_DIR_OUT);
+	gpio_pin_write(drv_data->gpio, CONFIG_CCS811_GPIO_WAKEUP_PIN_NUM, 0);
+
+	k_sleep(1);
+#endif
+
+	/* Switch device to application mode */
+	ret = switch_to_app_mode(drv_data->i2c);
+	if (ret)
+		return ret;
+
+	/* Check Hardware ID */
+	if (i2c_reg_read_byte(drv_data->i2c, CONFIG_CCS811_I2C_ADDR,
+			      CCS811_REG_HW_ID, &hw_id) < 0) {
+		SYS_LOG_ERR("Failed to read Hardware ID register");
+		return -EIO;
+	}
+
+	if (hw_id != CCS881_HW_ID) {
+		SYS_LOG_ERR("Hardware ID mismatch!");
+		return -EINVAL;
+	}
+
+	/* Set Measurement mode for 1 second */
+	if (i2c_reg_write_byte(drv_data->i2c, CONFIG_CCS811_I2C_ADDR,
+			       CCS811_REG_MEAS_MODE,
+			       CCS811_MODE_IAQ_1SEC) < 0) {
+		SYS_LOG_ERR("Failed to set Measurement mode");
+		return -EIO;
+	}
+
```

Comment:
It would be good to check the CCS811_STATUS  for errors after the drive mode is selected.

Reply:
Since drive mode is not user configurable, I didn't thought about implementing the error check. It won't hurt to add it though ;-)